editor/src/components/editor/global-shortcuts.tsx

```diff
@@ -954,3 +975,55 @@ function addCreateHoverInteractionActionToSwitchModeAction(
       ]
     : [switchModeAction]
 }
+
+function detectBestWrapperElement(
+  metadata: ElementInstanceMetadataMap,
+  elementPath: ElementPath,
+  makeUid: () => string,
+): WrapInView['whatToWrapWith'] {
+  const element = MetadataUtils.findElementByElementPath(metadata, elementPath)
+  if (
+    element == null ||
+    element.specialSizeMeasurements.parentFlexDirection == null ||
+    element.specialSizeMeasurements.parentLayoutSystem !== 'flex'
+  ) {
+    return 'default-empty-div'
+  }
+
+  const uid = makeUid()
+
+  // pragmatismus maximus, REVIEW: is there a better way to do this?
+  const style: Record<string, any> = {
+    display: 'flex',
+    flexDirection: element.specialSizeMeasurements.parentFlexDirection,
+    contain: 'layout',
+  }
+
+  if (
+    element.specialSizeMeasurements.parentFlexGap != null &&
+    element.specialSizeMeasurements.parentFlexGap !== 0
+  ) {
+    style.gap = element.specialSizeMeasurements.parentFlexGap
+  }
+
+  const props = jsxAttributesFromMap({
+    'data-uid': jsxAttributeValue(uid, emptyComments),
+    style: jsxAttributeValue(style, emptyComments),
+  })
+
+  if (element.specialSizeMeasurements.parentFlexDirection.startsWith('row')) {
+    return {
+      element: jsxElement('div', uid, props, []),
+      importsToAdd: {},
+    }
+  }
+
+  if (element.specialSizeMeasurements.parentFlexDirection.startsWith('col')) {
+    return {
+      element: jsxElement('div', uid, props, []),
+      importsToAdd: {},
+    }
+  }
+
+  return 'default-empty-div'
```

Comment:
You've already checked that there is a `parentFlexDirection`, meaning one of these two must be true, so I'd say go with an `if else` and bin off this redundant fallback case

Reply:
it turns out that both ifs can be removed (since `element.specialSizeMeasurements.parentFlexDirection` is already incorporated into `props` a few lines above this snippet)

Reply:
implemented in https://github.com/concrete-utopia/utopia/pull/3209/commits/c2eb0c2ee18d7b8a7b95370b2daa4fe077c31625